core/src/rust_types.rs

```diff
@@ -75,8 +75,17 @@ pub struct RustField {
     /// for the languages we generate code for.
     pub has_default: bool,
     /// Language-specific decorators assigned to a given field.
-    /// The keys are language names (e.g. SupportedLanguage::TypeScript), the values are decorators (e.g. readonly)
-    pub decorators: HashMap<SupportedLanguage, HashSet<String>>,
+    /// The keys are language names (e.g. SupportedLanguage::TypeScript), the values are field decorators (e.g. readonly)
+    pub decorators: HashMap<SupportedLanguage, BTreeSet<FieldDecorator>>,
+}
+
+/// A single decorator on a field in Rust code.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum FieldDecorator {
+    /// A boolean flag enabled by its existence as a decorator: for example, `readonly`.
+    Word(String),
+    /// A key-value pair, such as `type = "any"`.
+    NameValue(String, String),
```

Comment:
Recommend using `syn` types like `Ident` instead of strings here

Reply:
We can't do this because parsing is multi-threaded and `Ident` is not `Send`able.

Reply:
Why on god's green earth is `Ident` not `Send`